### ISSUE TITLE
%gui qt did not work and napari window did not pop up

### ISSUE THREAD

**Issue by rach-crc** [labsyspharm/scimap#121]
Dear Scimap developers,

I hope this message finds you well. I was trying to run the manul gating scipt on jupyter book, but %gui qt did not seem to work, as everytime I restart the kernel and run this cell, it always returnd a python3.10 file at the app bar, which cannot be opend or closed. No window of napari was shown. I am using Mac Studio, Apple M2 Ultra, macos version 14.5, SCIMAP  2.2.6. I downloaded the environment as conda create --name scimap python=3.10
conda activate scimap
pip install "scimap[napari]"
I also check my pyqt installation and it returned: (scimap) xxx-Mac-Studio ~ % pip install pyqt6
"Requirement already satisfied: pyqt6 in /opt/anaconda3/envs/scimap/lib/python3.10/site-packages (6.7.1)
Requirement already satisfied: PyQt6-sip<14,>=13.8 in /opt/anaconda3/envs/scimap/lib/python3.10/site-packages (from pyqt6)(13.8.0)
Requirement already satisfied: PyQt6-Qt6<6.8.0,>=6.7.0 in /opt/anaconda3/envs/scimap/lib/python3.10/site-packages (from pyqt6) (6.7.3)" However, when I was setting and setting the dependencies them in jupyter notebook, pyqt 6 -- NA (see the last bit of this issue), I don't know if that the problem that caused %gui qt to fall. I also checked whether if the problem from Napari, but running import napari and viewer = napari.Viewer() opend at a Napari window.
Could you please kindly help me check what is going on? I have attach a creenshot of my code to this message and return information below. 
<img width="984" alt="Screenshot 2024-11-29 at 22 10 44" src="https://github.com/user-attachments/assets/06b8e5eb-9833-44b1-a975-f3a21ae3d207">

Thanks.
Here is what returns from the code:
Running SCIMAP  2.2.6
Current date and time = 2024-11-29 21:29:27.906248
-----
anndata     0.10.9
scanpy      1.10.4
-----
OpenGL                      3.1.7
PIL                         11.0.0
PyQt6                       NA
appdirs                     1.4.4
appnope                     0.1.4
asciitree                   NA
asttokens                   NA
attr                        24.2.0
attrs                       24.2.0
babel                       2.16.0
backports                   NA
cachey                      0.2.1
certifi                     2024.08.30
charset_normalizer          3.4.0
cloudpickle                 3.1.0
colorama                    0.4.6
combat                      NA
comm                        0.2.2
cycler                      0.12.1
cython_runtime              NA
dask                        2023.12.1
dateutil                    2.9.0.post0
debugpy                     1.8.9
decorator                   5.1.1
docstring_parser            NA
exceptiongroup              1.2.2
executing                   2.1.0
fast_histogram              0.14
fasteners                   0.19
gensim                      4.3.3
h5py                        3.12.1
heapdict                    NA
hsluv                       5.0.4
idna                        3.10
igraph                      0.11.8
ipykernel                   6.29.5
ipywidgets                  8.1.5
jaraco                      NA
jedi                        0.19.2
jinja2                      3.1.4
joblib                      1.4.2
jsonschema                  4.23.0
jsonschema_specifications   NA
kiwisolver                  1.4.7
lazy_loader                 0.4
legacy_api_wrap             NA
leidenalg                   0.10.2
llvmlite                    0.41.1
magicgui                    0.9.1
markupsafe                  3.0.2
matplotlib                  3.9.2
matplotlib_inline           0.1.7
more_itertools              10.3.0
mpl_scatter_density         0.7
mpl_toolkits                NA
mpmath                      1.3.0
msgpack                     1.1.0
napari                      0.4.19.post1
napari_plugin_engine        0.2.0
natsort                     8.4.0
networkx                    3.4.2
npe2                        0.7.7
numba                       0.58.1
numcodecs                   0.13.1
numpy                       1.26.4
numpydoc                    1.8.0
packaging                   24.2
pandas                      2.2.3
parso                       0.8.4
patsy                       1.0.1
pkg_resources               NA
platformdirs                4.3.6
plotly                      5.24.1
pooch                       v1.8.2
prompt_toolkit              3.0.48
psutil                      6.1.0
psygnal                     0.11.1
pure_eval                   0.2.3
pydantic                    2.10.2
pydev_ipython               NA
pydevconsole                NA
pydevd                      3.2.3
pydevd_file_utils           NA
pydevd_plugins              NA
pydevd_tracing              NA
pygments                    2.18.0
pynndescent                 0.5.13
pyparsing                   3.2.0
pytz                        2024.2
qtpy                        2.4.2
referencing                 NA
requests                    2.32.3
rpds                        NA
scimap                      2.2.6
scipy                       1.12.0
seaborn                     0.13.2
session_info                1.0.0
shapely                     2.0.6
six                         1.16.0
skimage                     0.24.0
sklearn                     1.5.2
smart_open                  7.0.5
sphinxcontrib               NA
stack_data                  0.6.3
statsmodels                 0.14.4
tblib                       3.0.0
tenacity                    NA
texttable                   1.7.0
threadpoolctl               3.5.0
tifffile                    2023.12.9
tlz                         1.0.0
toolz                       1.0.0
tornado                     6.4.2
tqdm                        4.67.1
traitlets                   5.14.3
typing_extensions           NA
umap                        0.5.7
urllib3                     2.2.3
vispy                       0.14.3
wcwidth                     0.2.13
wrapt                       1.17.0
yaml                        6.0.2
zarr                        2.10.3
zipp                        NA
zmq                         26.2.0
zoneinfo                    NA
-----
IPython             8.30.0
jupyter_client      8.6.3
jupyter_core        5.7.2
-----
Python 3.10.15 (main, Oct  3 2024, 02:24:49) [Clang 14.0.6 ]
macOS-14.5-arm64-arm-64bit
-----
Session information updated at 2024-11-29 21:29
/opt/anaconda3/envs/scimap/lib/python3.10/site-packages/session_info/main.py:213: UserWarning:

The '__version__' attribute is deprecated and will be removed in MarkupSafe 3.1. Use feature detection, or `importlib.metadata.version("markupsafe")`, instead.

**Comment by ajitjohnson** [comment]
If Napari is working, it likely has to do with the data. Can you describe what data are you using, how was it imported into scimap and have you run other scimap function on the data?

could you also run the old gating function `https://scimap.xyz/Functions/pl/gate_finder/` just to check if it a bug with the new one?

Best,
Ajit.

**Comment by rach-crc** [comment]
Thank you for the reply. Turns out it's my own problem. Everything now worked. But I have two other questions regarding manual gating and I would be very appreciated if you can help me with them. 1. after I run sm.pl.gate_finder it noticed me that this function may be replaced by sm.pl.napriGater in the future, but based on the napariGater function instruction: https://scimap.xyz/Functions/pl/napariGater/ it does not have from_gate=6, to_gate=8, increment=0.1 as in gate_finder, hence I wonder if the two functions here are exactly the same. 2. I wonder if there is a way to check the intensity (e.g., a histogram) of individual markers for selecting the range for gating, for example, if I set from_gate=6, to_gate=8 it did not work for my image as they are quite dim and it will take time to try changing the gate number to find the correct range (around 2 eventually), so I think if there is a histogram of the distribution of marker intensity would facilitate the speed of find the correct range of gates and then adjust to find the optimal one.

Have a nice week.

**Comment by rach-crc** [comment]
> If Napari is working, it likely has to do with the data. Can you describe what data are you using, how was it imported into scimap and have you run other scimap function on the data?
> 
> could you also run the old gating function `https://scimap.xyz/Functions/pl/gate_finder/` just to check if it a bug with the new one?
> 
> Best, Ajit.

Thank you for the reply. Turns out it's my own problem. Everything now worked. But I have two other questions regarding manual gating and I would be very appreciated if you can help me with them. 1. after I run sm.pl.gate_finder it noticed me that this function may be replaced by sm.pl.napriGater in the future, but based on the napariGater function instruction: https://scimap.xyz/Functions/pl/napariGater/ it does not have from_gate=6, to_gate=8, increment=0.1 as in gate_finder, hence I wonder if the two functions here are exactly the same. 2. I wonder if there is a way to check the intensity (e.g., a histogram) of individual markers for selecting the range for gating, for example, if I set from_gate=6, to_gate=8 it did not work for my image as they are quite dim and it will take time to try changing the gate number to find the correct range (around 2 eventually), so I think if there is a histogram of the distribution of marker intensity would facilitate the speed of find the correct range of gates and then adjust to find the optimal one.

Have a nice week.

**Comment by ajitjohnson** [comment]
I'm glad to see you've gotten it to work! Just a heads-up: `gate_finder()` will be phased out, so I recommend switching to `napariGater()`. With `napariGater()`, there’s no need to specify parameters like `from_gate` and `to_gate`—it automatically loads the full range and initializes with an estimated gate, which you can then fine-tune as needed.

**Comment by rach-crc** [comment]
> I'm glad to see you've gotten it to work! Just a heads-up: `gate_finder()` will be phased out, so I recommend switching to `napariGater()`. With `napariGater()`, there’s no need to specify parameters like `from_gate` and `to_gate`—it automatically loads the full range and initializes with an estimated gate, which you can then fine-tune as needed.

Thank you so much, Ajit. napariGater() is working quite well, thank you for the hard work.
